Building from source on Amazon Linux for ARM does not work after rel 1.0.20

Building with rel 1.0.20 works fine
When Building with master I get the following error

```
sh-5.2$ make -j
Making all in third_party/luajit
make[1]: Entering directory '/home/ssm-user/sysbench/third_party/luajit'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/ssm-user/sysbench/third_party/luajit'
Making all in third_party/concurrency_kit
make[1]: Entering directory '/home/ssm-user/sysbench/third_party/concurrency_kit'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/ssm-user/sysbench/third_party/concurrency_kit'
Making all in src
make[1]: Entering directory '/home/ssm-user/sysbench/src'
Making all in drivers
make[2]: Entering directory '/home/ssm-user/sysbench/src/drivers'
Making all in mysql
make[3]: Entering directory '/home/ssm-user/sysbench/src/drivers/mysql'
gcc -DHAVE_CONFIG_H -I. -I../../../config  -I/usr/include/mysql -I/usr/include/mysql/mysql -I../../../src -I/home/ssm-user/sysbench/third_party/luajit/inc -I/home/ssm-user/sysbench/third_party/concurrency_kit/include -D_GNU_SOURCE  -DCK_MD_CACHELINE=128 -Wall -Wextra -Wpointer-arith -Wbad-function-cast -Wstrict-prototypes -Wnested-externs -Wno-format-zero-length -Wundef -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Wcast-align -Wvla    -O3 -funroll-loops -ggdb3  -MT libsbmysql_a-drv_mysql.o -MD -MP -MF .deps/libsbmysql_a-drv_mysql.Tpo -c -o libsbmysql_a-drv_mysql.o `test -f 'drv_mysql.c' || echo './'`drv_mysql.c
drv_mysql.c: In function ‘mysql_drv_real_connect’:
drv_mysql.c:420:24: error: ‘MYSQL_OPT_COMPRESSION_ALGORITHMS’ undeclared (first use in this function)
  420 |     mysql_options(con, MYSQL_OPT_COMPRESSION_ALGORITHMS, args.compression_alg);
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drv_mysql.c:420:24: note: each undeclared identifier is reported only once for each function it appears in
make[3]: *** [Makefile:440: libsbmysql_a-drv_mysql.o] Error 1
make[3]: Leaving directory '/home/ssm-user/sysbench/src/drivers/mysql'
make[2]: *** [Makefile:412: all-recursive] Error 1
make[2]: Leaving directory '/home/ssm-user/sysbench/src/drivers'
make[1]: *** [Makefile:591: all-recursive] Error 1
make[1]: Leaving directory '/home/ssm-user/sysbench/src'
make: *** [Makefile:481: all-recursive] Error 1
```

uname -r: 6.1.34-59.116.amzn2023.aarch64

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -24,3 +24,28 @@ jobs:
       - name: Test
         run: make test
 
+  build_mariadb:
+    runs-on: ubuntu-22.04
+    name: Build with MariaDB
+    steps:
+      - name: Setup MariaDB Repo
+        run: |
+          sudo apt-get install software-properties-common
+          sudo apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 0xF1656F24C74CD1D8
+          sudo add-apt-repository 'deb [arch=amd64] https://ftp.nluug.nl/db/mariadb/repo/11.0/ubuntu jammy main'
+      - name: Setup MariaDB Libs
+        run: sudo apt install libmariadb-dev libmariadb-dev-compat
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Autogen
+        run: ./autogen.sh
+      - name: Configure
+        run: ./configure --with-mysql --with-pgsql
+      - name: Build
+        run: make
+      - name: MariaDB version
+        run: mariadb_config --version
+      - name: Sysbench version
+        run: ./src/sysbench --version
+      - name: Test
+        run: make test

--- a/src/drivers/mysql/drv_mysql.c
+++ b/src/drivers/mysql/drv_mysql.c
@@ -109,7 +109,9 @@ typedef struct
   const char         *ssl_ca;
   const char         *ssl_cipher;
   unsigned char      use_compression;
+#ifdef MYSQL_OPT_COMPRESSION_ALGORITHMS
   const char         *compression_alg;
+#endif
   unsigned char      debug;
   sb_list_t          *ignored_errors;
   unsigned int       dry_run;
@@ -337,7 +339,9 @@ int mysql_drv_init(void)
 #endif
 
   args.use_compression = sb_get_value_flag("mysql-compression");
+#ifdef MYSQL_OPT_COMPRESSION_ALGORITHMS
   args.compression_alg = sb_get_value_string("mysql-compression-algorithms");
+#endif
 
   args.debug = sb_get_value_flag("mysql-debug");
   if (args.debug)
@@ -416,8 +420,10 @@ static int mysql_drv_real_connect(db_mysql_conn_t *db_mysql_con)
     DEBUG("mysql_options(%p, %s, %s)",con, "MYSQL_OPT_COMPRESS", "NULL");
     mysql_options(con, MYSQL_OPT_COMPRESS, NULL);
 
+#ifdef MYSQL_OPT_COMPRESSION_ALGORITHMS
     DEBUG("mysql_options(%p, %s, %s)",con, "MYSQL_OPT_COMPRESSION_ALGORITHMS", args.compression_alg);
     mysql_options(con, MYSQL_OPT_COMPRESSION_ALGORITHMS, args.compression_alg);
+#endif
   }
 
   DEBUG("mysql_real_connect(%p, \"%s\", \"%s\", \"%s\", \"%s\", %u, \"%s\", %s)",

--- a/tests/t/help_drv_mysql.t
+++ b/tests/t/help_drv_mysql.t
@@ -8,7 +8,7 @@ Skip test if the MySQL driver is not available.
   $ sysbench --help | grep -- '--db-driver'
     --db-driver=STRING  specifies database driver to use ('help' to get list of available drivers) [mysql]
 
-  $ sysbench --help | sed -n '/mysql options:/,/^$/p'
+  $ sysbench --help | sed -n '/mysql options:/,/^$/p' | grep -v 'mysql-ssl[=\[]'
   mysql options:
     --mysql-host=[LIST,...]               MySQL server host [localhost]
     --mysql-port=[LIST,...]               MySQL server port [3306]
@@ -16,7 +16,6 @@ Skip test if the MySQL driver is not available.
     --mysql-user=STRING                   MySQL user [sbtest]
     --mysql-password=STRING               MySQL password []
     --mysql-db=STRING                     MySQL database name [sbtest]
-    --mysql-ssl=STRING                    SSL mode. This accepts the same values as the --ssl-mode option in the MySQL client utilities. Disabled by default [disabled]
     --mysql-ssl-key=STRING                path name of the client private key file
     --mysql-ssl-ca=STRING                 path name of the CA file
     --mysql-ssl-cert=STRING               path name of the client public key certificate file

